sintr_analysis_server_example/test_driver.dart

```diff
@@ -85,3 +69,55 @@ main(List<String> args) async {
   var reduced = completionReductionMerge(reduced1, reduced2);
   print(reduced);
 }
+
+Future<Map> loadSessionInfo(String path) async {
+  var index = path.indexOf('-', path.lastIndexOf('/'));
+  var priPath = '${path.substring(0, index)}-PRI${path.substring(index + 1)}';
+  var f = new io.File(priPath);
+
+  var firstLine;
+  LogItemProcessor proc = new LogItemProcessor((line) {
+    if (firstLine == null) firstLine = line;
+  });
+  await for (String ln
+      in f.openRead().transform(UTF8.decoder).transform(new LineSplitter())) {
+    proc.addRawLine(ln);
+    processMessages(proc);
+  }
+  proc.close();
+  processMessages(proc);
+
+  var data = firstLine.split(':');
+  return {
+    'uuid': data[2],
+    'clientId': data[3],
+    'clientVersion': data[4],
+    'serverVersion': data[5],
+    'sdkVersion': data[6]
+  };
+}
+
+void processMessages(LogItemProcessor proc, [List<String> extracted]) {
```

Comment:
Yes, that makes sense. Per discussion, in a subsequent PR I'll extract the LogProcessor into one library and the PRI file extraction into another.